### ISSUE TITLE
MD029/ol-prefix: an extra space is needed for code fences in lists with more than 10 elements

### ISSUE THREAD

**Issue by zachliu** [DavidAnson/markdownlint#1345]
I created a list with more than 10 entries. All of them have a code block in it.

See example:

``````markdown
# Test

1. ...
2. ...
   1. three spaces here
   2. three spaces here
3. ...:

   ```bash
   three spaces here
   ```

4. ...:

   ```bash
   three spaces here
   ```

5. ...:

   ```bash
   three spaces here
   ```

6. ...:

   ```bash
   three spaces here
   ```

7. ...:

   ```bash
   three spaces here
   ```

8. ...:

   ```bash
   three spaces here
   ```

9. ...:

   ```bash
   three spaces here
   ```

10. ...:

    ```bash
    double tab (four spaces) here
    ```

11. ...:

    ```bash
    double tab (four spaces) here
    ```

12. ...:

    ```bash
    double tab (four spaces) here
    ```
``````

I discovered that I had to use `4` spaces instead of `3` (mentioned in https://github.com/DavidAnson/markdownlint/issues/38 and https://github.com/DavidAnson/markdownlint/issues/125) for code fences if their index number is >= 10 otherwise I get `MD029/ol-prefix Ordered list item prefix [Expected: 1; Actual: 11; Style: 1/2/3]` on items 11, 12, ... etc (interestingly not on item 10).

**Comment by DavidAnson** [comment]
Your example above produces no warnings from markdownlint as-is. The code fences in list items 10 and above need four spaces of indent because that's what corresponds to the first character after the previous list item marker. Specifically, "9. " is only three characters whereas "10. " is four, and thus the change. Per the CommonMark specification, using fewer spaces than required to nest the code fence within a list item causes the next list item marker to begin a new list. For example, insufficiently indenting the list in item 10 causes a warning for item 11 because it creates a new list that doesn't start with 1. So far as I can tell, everything here is correct and operating as intended.

**Comment by zachliu** [comment]
sorry for the confusion. the example was provided so that it produced no warnings

![2024-09-03_19-44](https://github.com/user-attachments/assets/044c6007-8ab2-4f69-87aa-1139ac6d77de)

had i not used 4 spaces for 10, 11, and 12, i would get the warnings

**Comment by zachliu** [comment]
> The code fences in list items 10 and above need four spaces of indent because that's what corresponds to the first character after the previous list item marker. Specifically, "9. " is only three characters whereas "10. " is four, and thus the change.

i see the point :+1: 
the number of spaces needs is the same as the indexes themselves: `9. ` needs three, `10. ` needs four, `100. ` needs five

**Comment by DavidAnson** [comment]
Right, without at least that many spaces, the code fence is not part of the list item as can be seen in this example where the middle fence has 3 instead of 4 spaces of indent: https://dlaa.me/markdownlint/#%25m%23%20Test%0A%0A9.%20...%3A%0A%0A%20%20%20%60%60%60bash%0A%20%20%20three%0A%20%20%20%60%60%60%0A%0A10.%20...%3A%0A%0A%20%20%20%60%60%60bash%0A%20%20%20three%0A%20%20%20%60%60%60%0A%0A11.%20...%3A%0A%0A%20%20%20%20%60%60%60bash%0A%20%20%20%20four%0A%20%20%20%20%60%60%60%0A
